### ISSUE TITLE
Command "sudo iotedge check" gives false feedback if config.yaml uses DPS connection

### ISSUE THREAD

**Issue by JeeWeetje** [Azure/iotedge#1232]
Since version 1.0.7 there is the ability to run the command
`sudo iotedge check`

This checks the connection string as second item of the checklist. But when using a Azure Device Provision Service connection configuration the check shows a false failed check on the device connection string configuration.

```
√ config.yaml is well-formed
× config.yaml has well-formed connection string
    Device is not using manual provisioning, so Azure IoT Hub hostname needs to be specified with --iothub-hostname
```

## Expected Behavior
Expected output would be a validation of the DPS connection configuration (which includes more than one step of configuration) showing a positive check message is checks pass

```
√ config.yaml is well-formed
√ config.yaml has well-formed DPS connection string, Azure IoT Hub host name resolved from DPS.
```

## Current Behavior
The command `sudo iotedge check` provides a hit how to use the check option with non manual connection string configurations.

`Device is not using manual provisioning, so Azure IoT Hub hostname needs to be specified with --iothub-hostname`

When applying the --iothub-hostname option:

`sudo iotedge check --iothub-hostname <name of your iothub>.azure-devices.net`

It only checks if you did use the correct URI schema. It does not check if the provided hostname of the IoT Hub is actually the IoT Hub where the device should be registered. The check `√ config.yaml has well-formed connection string` will be green.

If you apply a bogus URI like whatever.com the steps for connectivity checks will fail while step 2 will check as ok (`√ config.yaml has well-formed connection string.`). 

## Steps to Reproduce
0. Provision a VM or configure a real machine with TPM2.0 support 
1. Provision your Azure IoT Edge device 
2. Install Azure IoT Edge runtime version 1.0.7
3. Run sudo iotedge check
4. Run sudo iotedge check --iothub-hostname whatever.com

## Context (Environment)
Test machines:
Advantech UNO-1372G
Advantech UNO 2372G
Advantech UNO 2484G

### Device (Host) Operating System
Ubuntu Server 18.04 LTS, up part with package updates and security updates (May 20, 2019).

### Architecture
amd64

### Container Operating System
Linux containers

### Runtime Versions
Moby Engine 3.0.5
Moby CLI 3.0.5
Azure IoT Edge 1.0.7
Libiothsm-std 1.0.7

#### iotedged
iotedge 1.0.7 (f455ae2cd66db716c4bd8e7aaa7984ce092481c0)

#### Edge Agent
1.0.7 (deployment manifest hardcoded)

#### Edge Hub
1.0.7 (deployment manifest hardcoded)

#### Docker
`Client:
 Version:           3.0.5
 API version:       1.40
 Go version:        go1.12.1
 Git commit:        ba9934d4
 Built:             Thu Apr 18 22:01:41 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          3.0.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.1
  Git commit:       dbe4a30
  Built:            Thu Apr 18 22:07:58 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.5
  GitCommit:        bb71b10fd8f58240ca47fbb579b9d1028eea7c84
 runc:
  Version:          1.0.0-rc6+dev
  GitCommit:        2b18fe1d885ee5083ef9f0838fee39b62d653e30
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683`

## Logs
![iotedge-check-logs](https://user-images.githubusercontent.com/693760/58050383-7c53a480-7b4f-11e9-82a6-178f50bb3097.png)

## Additional Information
Provisioned a real TPM 2.0 enabled device (3 different models). All give the same check outcome.

**Comment by varunpuranik** [comment]
@JeeWeetje - Thanks for the issue. 
To make sure I understand your issue correctly - When config.yaml is configured in DPS mode, and you provide IoTHub hostname to iotedge check command, it does not check if the IoTHub url is valid, and some later checks fail? 
@veyalla @arsing

**Comment by arsing** [comment]
It's expected that the command succeeds even if you give it an invalid value for `--iothub-hostname`. The tool does not try to actually authenticate with the IoT Hub in any way (nor could it, since that's `iotedged`'s job). The hostname is only used to test for connectivity of the three protocols.

**Comment by arsing** [comment]
Confirmed with @mrohera that there's no validation that `iotedge` can do, other than a minimal "ensure that the DPS endpoint resolves and is reachable over HTTPS". We can add that as a step, but the full DPS enrolment validation as mentioned in the OP is not possible.

**Comment by JeeWeetje** [comment]
@arsing Is it also possible in that the scope id gets validated at the DPS endpoint? The combination of the DPS end point and the scope id will provide at least prove there is an DPS active for registrations.

I understand that the check if for the enrollment is a lot of work, but it is also used in the example documentation for registering a device. See https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/provisioning/Samples/device/TpmSample/Program.cs

The reason for asking can be shared offline. @veyalla has my contact details.

**Comment by arsing** [comment]
>>[...] other than a minimal "ensure that the DPS endpoint resolves and is reachable over HTTPS". We can add that as a step, [...]

#1268 added a check to resolve and perform a TLS handshake with the DPS endpoint.


>I understand that the check if for the enrollment is a lot of work, but it is also used in the example documentation for registering a device.

I don't think it is in scope for iotedge-check, especially since the context of iotedge-check is to run on a device that may already be running a real IoT Edge deployment with a real DPS registration.

If the intention is to discover issues with the DPS enrolment, you can check iotedged logs to discover the cause.

Note that this has parity with the corresponding check for manual provisioning, since that also does *not* authenticate with the IoT Hub to ensure that the shared access key is a valid key. It is possible to give a wrong key in the connection string and only discover this from iotedged logs.

**Comment by JeeWeetje** [comment]
@arsing Thank you for the scoping of iotedge check. And thanks for adding the DPS endpoint check. 

This is in scope for version 1.0.9 release right? Eager to test it on a RC version.

**Comment by crpietschmann** [comment]
When might we see this bug fix and v1.0.9 released?
